Xamarin.Forms.Core/BindableObject.cs

```diff
@@ -486,7 +486,7 @@ BindablePropertyContext GetContext(BindableProperty property)
 			for (var i = 0; i < properties.Count; i++)
 			{
 				BindablePropertyContext context = properties[i];
-				if (ReferenceEquals(context.Property, property))
+				if (context != null && ReferenceEquals(context.Property, property))
```

Comment:
@rotorgames looks like you can reproduce your issue in a reliable way, we'll need a test. Attach it to one of your bug reports (I'm still suspecting threading issues).

It it NOT supported to make changes to BP from other threads than the main one.